handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -614,21 +614,43 @@ public final SSLEngineResult wrap(
                 }
             }
 
-            if (dst.remaining() < MAX_ENCRYPTED_PACKET_LENGTH) {
+            int endOffset = offset + length;
+            int srcsLen = 0;
+
+            for (int i = offset; i < endOffset; ++i) {
+                final ByteBuffer src = srcs[i];
+                if (src == null) {
+                    throw new IllegalArgumentException("srcs[" + i + "] is null");
+                }
+                if (srcsLen == MAX_PLAINTEXT_LENGTH) {
+                    continue;
+                }
+
+                srcsLen += src.remaining();
+                if (srcsLen > MAX_PLAINTEXT_LENGTH) {
+                    // If srcLen > MAX_PLAINTEXT_LENGTH just set it to MAX_PLAINTEXT_LENGTH.
+                    // This also help us to guard against overflow.
+                    // We not break out here as we still need to check for null entries in srcs[].
+                    srcsLen = MAX_PLAINTEXT_LENGTH;
+                }
+            }
+
+            int maxEncryptedLen = calculateOutNetBufSize(srcsLen);
+
+            if (MAX_ENCRYPTED_PACKET_LENGTH < maxEncryptedLen) {
```

Comment:
`calculateOutNetBufSize` already takes `MAX_ENCRYPTED_PACKET_LENGTH` into account so this condition isn't necessary.